src/wallet/scriptpubkeyman.cpp

```diff
@@ -2732,10 +2734,13 @@ void DescriptorScriptPubKeyMan::UpdateWalletDescriptor(WalletDescriptor& descrip
         throw std::runtime_error(std::string(__func__) + ": " + error);
     }
 
+    bool signal_birth_time = descriptor.creation_time < m_wallet_descriptor.creation_time;
     m_map_pubkeys.clear();
     m_map_script_pub_keys.clear();
     m_max_cached_index = -1;
     m_wallet_descriptor = descriptor;
+
+    if (signal_birth_time) NotifyFirstKeyTimeChanged(this, m_wallet_descriptor.creation_time);
```

Comment:
To save some extra processing, because they are performed at different levels.
At the spkm level (here), we are at the lowest one. So, before emitting an event that will require to be processed by all the listeners (the wallets and/or other registered objects), we can just check if the updated descriptor timestamp is lower than the existent one.